### PR TITLE
DHS: Clean various warnings, violations of ISO C and undefined behavior.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: tools/dhs/include/dhs_patch/dhs_patch_compat.h [hunk 1]
```diff
@@ -131,4 +164,4 @@ dhs_a9_compat_s dhs_a9_compat_49 = {
 
 #endif /*DHS_PATCH_COMPAT_DATA*/
 
-#endif /*__DHS_PATCH_COMPAT_H*/
+#endif /*DHS_PATCH_COMPAT_H*/
```
Comment (b1l1s) [inline tools/dhs/include/dhs_patch/dhs_patch_compat.h:167]: I agree with all changes except for these.
Any specific reason why you want to move the file declarations to compat?

Comment (ghost) [inline tools/dhs/include/dhs_patch/dhs_patch_compat.h:167]: file.h and dhs_patch_compat.h have typedefs that always cause a redefinition of the IFILE type. I copied it to dhs_patch_compat.h to fix.

Comment (b1l1s) [inline tools/dhs/include/dhs_patch/dhs_patch_compat.h:167]: Noted.

**Comment by b1l1s** [comment]
Also have this been tested? I got my hands tied right now so I can't spend too much time checking what the problem is, but I keep getting communication error. I'll merge this once I'm more free.

*Also at a glance the process name copying will fail since it stops copying at '\0'. You may want to set the cmd.namelo and cmd.namehi to zeroes first.

**Comment by b1l1s** [comment]
Sorry but I'm going to have to close this. With me rebasing dhs and the bug in the commit I can't pull this.

**Comment by Syphurith** [comment]
@b1l1s Try cherrypick the fix-warning commit if there are still such warnings and it applies.

**Comment by b1l1s** [comment]
O_o I'm not gonna cherrypick through "warnings" PR. Not gonna happen. If you want to do PRs like these just make sure it compiles fine, tested (you bet your ass I'm gonna test it first too) and I'll happily merge. Just don't make me do the dirty work of fixing your PR. /endrant

**Comment by Syphurith** [comment]
@b1l1s I thought it could compile and does fix before your reply.`スクリーンショット`=`ScreenShot`

**Comment by b1l1s** [comment]
92a862d643abe45b5b6974fc07801d96369901eb July 5th <- screenshot
096e608f88c65218b7e3b2e338ab75446f7b3495 July 7th <- my last commit before this
bc32adf300cdfb70b16b582f11a648d42329511e July 8th <- this commit

**Comment by Syphurith** [comment]
@b1l1s Well. it might be dealt later. I've no question about this then.

**Pull request closed without merging**